In [10]:
#import libraries
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [11]:
# load the dataset
df=pd.read_csv(r'C:\Users\E6440\Desktop\Code_Alpha_internship\SuperMarket Analysis.csv')

In [12]:
# display the top five rows of the dataset
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             Product line  Unit price  Quantity   Tax 5%     Sales       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

          Time      Payment    cogs  gross margin percentage  gross income  \
0   1:08:00 PM      Ewallet  522.83                 4.761905       26.1415   
1  10:29:00 AM         Cash   76.40                 4.761905        3.8200   
2   1:23:00 PM  Credit card  324.31                 4.761905       16.2155   
3   8:33:00 PM      Ewallet  465.76                 4.761905       23.2880   
4  10:37:00 AM      Ewallet  604.17                 4.761905       30.2085   

   Rating  
0     9.1  
1     9.6  
2     7.4  
3     8.4  
4     5.3

In [13]:
# display the five bottom rows
df.tail()

Invoice ID Branch       City Customer type  Gender         Product line  \
995  233-67-5758   Giza  Naypyitaw        Normal    Male    Health and beauty   
996  303-96-2227  Cairo   Mandalay        Normal  Female   Home and lifestyle   
997  727-02-1313   Alex     Yangon        Member    Male   Food and beverages   
998  347-56-2442   Alex     Yangon        Normal    Male   Home and lifestyle   
999  849-09-3807   Alex     Yangon        Member  Female  Fashion accessories   

     Unit price  Quantity   Tax 5%      Sales       Date        Time  Payment  \
995       40.35         1   2.0175    42.3675  1/29/2019  1:46:00 PM  Ewallet   
996       97.38        10  48.6900  1022.4900   3/2/2019  5:16:00 PM  Ewallet   
997       31.84         1   1.5920    33.4320   2/9/2019  1:22:00 PM     Cash   
998       65.82         1   3.2910    69.1110  2/22/2019  3:33:00 PM     Cash   
999       88.34         7  30.9190   649.2990  2/18/2019  1:28:00 PM     Cash   

       cogs  gross margin percentage  gross income  Rating  
995   40.35                 4.761905        2.0175     6.2  
996  973.80                 4.761905       48.6900     4.4  
997   31.84                 4.761905        1.5920     7.7  
998   65.82                 4.761905        3.2910     4.1  
999  618.38                 4.761905       30.9190     6.6

In [14]:
# check the columns of the dataset
df.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Sales', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

In [15]:
# check the datatypes of the dataset
df.dtypes

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Sales                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

In [16]:
df.describe()

Unit price     Quantity       Tax 5%        Sales        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count              1000.000000   1000.000000  1000.00000  
mean                  4.761905     15.379369     6.97270  
std                   0.000000     11.708825     1.71858  
min                   4.761905      0.508500     4.00000  
25%                   4.761905      5.924875     5.50000  
50%                   4.761905     12.088000     7.00000  
75%                   4.761905     22.445250     8.50000  
max                   4.761905     49.650000    10.00000

In [17]:
# check the null values in the dataset
df.isnull().sum()

Invoice ID                 0
Branch                     0
City                       0
Customer type              0
Gender                     0
Product line               0
Unit price                 0
Quantity                   0
Tax 5%                     0
Sales                      0
Date                       0
Time                       0
Payment                    0
cogs                       0
gross margin percentage    0
gross income               0
Rating                     0
dtype: int64

### Fix Date & Time Mandatory

In [18]:
df['Date'] = pd.to_datetime(df['Date'])
df['Time'] = pd.to_datetime(df['Time'])


C:\Users\E6440\AppData\Local\Temp\ipykernel_3284\596691792.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Time'] = pd.to_datetime(df['Time'])


In [28]:
df.head(2)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   

             Product line  Unit price  Quantity   Tax 5%     Sales       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   

                 Time  Payment    cogs  gross margin percentage  gross income  \
0 2026-02-04 13:08:00  Ewallet  522.83                 4.761905       26.1415   
1 2026-02-04 10:29:00     Cash   76.40                 4.761905        3.8200   

   Rating  Month       Day  Hour  
0     9.1      1  Saturday    13  
1     9.6      3    Friday    10

### Feature Engineering

In [19]:
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day_name()
df['Hour'] = df['Time'].dt.hour


In [20]:
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             Product line  Unit price  Quantity   Tax 5%     Sales       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

                 Time      Payment    cogs  gross margin percentage  \
0 2026-02-04 13:08:00      Ewallet  522.83                 4.761905   
1 2026-02-04 10:29:00         Cash   76.40                 4.761905   
2 2026-02-04 13:23:00  Credit card  324.31                 4.761905   
3 2026-02-04 20:33:00      Ewallet  465.76                 4.761905   
4 2026-02-04 10:37:00      Ewallet  604.17                 4.761905   

   gross income  Rating  Month       Day  Hour  
0       26.1415     9.1      1  Saturday    13  
1        3.8200     9.6      3    Friday    10  
2       16.2155     7.4      3    Sunday    13  
3       23.2880     8.4      1    Sunday    20  
4       30.2085     5.3      2    Friday    10

## Key Business Questions

1. Which product line has the highest sales?
2. Does gender affect total sales?
3. Which city performs best in terms of sales?
4. What payment method is most frequently used?


### EDA (Exploratary Data Analysis)

In [21]:
df.groupby('City')['Sales'].sum().sort_values(ascending=False)

City
Naypyitaw    110568.7065
Yangon       106200.3705
Mandalay     106197.6720
Name: Sales, dtype: float64

In [22]:
df.groupby('Branch')['Sales'].sum()

Branch
Alex     106200.3705
Cairo    106197.6720
Giza     110568.7065
Name: Sales, dtype: float64

In [23]:
df.groupby('Product line')['gross income'].sum().sort_values(ascending=False)

Product line
Food and beverages        2673.5640
Sports and travel         2624.8965
Electronic accessories    2587.5015
Fashion accessories       2585.9950
Home and lifestyle        2564.8530
Health and beauty         2342.5590
Name: gross income, dtype: float64

## Gross Income Analysis by Product Line

- **Food and beverages** have the highest gross income, while **Health and beauty** has the lowest.
- Other lines like **Sports and travel** and **Electronic accessories** also contribute significantly.
- Shows that **high revenue doesn’t always mean high quantity sold**.


In [24]:
df.groupby('Product line')['Quantity'].sum()

Product line
Electronic accessories    971
Fashion accessories       902
Food and beverages        952
Health and beauty         854
Home and lifestyle        911
Sports and travel         920
Name: Quantity, dtype: int64

###  Quantity Sold by Product Line

- **Electronic Accessories** – 971 units (highest sold)
- **Food and Beverages** – 952 units
- **Sports and Travel** – 920 units
- **Health and Beauty** – 854 units (lowest sold)


In [25]:
df.groupby('Customer type')['Sales'].sum()

Customer type
Member    189694.764
Normal    133271.985
Name: Sales, dtype: float64

###  Sales by Customer Type

- **Member** – 189,694.76 (highest revenue contributor)
- **Normal** – 133,271.99


In [26]:
df.groupby('Gender')['Sales'].sum()

Gender
Female    194671.8375
Male      128294.9115
Name: Sales, dtype: float64

###  Sales by Gender

- **Female** – 194,671.84 (highest revenue contributor)
- **Male** – 128,294.91


In [27]:
df['Payment'].value_counts()

Payment
Ewallet        345
Cash           344
Credit card    311
Name: count, dtype: int64

the analysis tells us the most of the customers payed the payment through the Ewallet which is 345 and second is cash customers pay the payment through the cash 344 we can say that the customers pay through the Ewallet and cash